ext/date/php_date.c

```diff
@@ -3318,11 +3337,14 @@ PHP_FUNCTION(date_timestamp_get)
 	timelib_update_ts(dateobj->time, NULL);
 
 	timestamp = timelib_date_to_int(dateobj->time, &error);
+
+	/* Seconds since epoch must fit in a zend_long */
 	if (error) {
+		/* TODO Add warning? */
```

Comment:
I'm thinking a ValueException or something is OK here, as the function/method *specifically* asks the code to return an integer since the epoch.